src/views/components/building-address/building-address-input-fields.njk

```diff
@@ -0,0 +1,75 @@
+{% from "hmpo-text/macro.njk" import hmpoText %}
+{%- from "govuk/components/hint/macro.njk" import govukHint %}
+{%- from "govuk/components/label/macro.njk" import govukLabel %}
+{% from "./macro.njk" import customInput %}
+
+{%- set args = {
+  id: "buildingAddress",
+  label: { text: "Building address" },
+  hint: { text: "Fill in at least 1 field in this section" },
+  wrapper: {
+    classes: "govuk-inset-text govuk-!-margin-top-0"
+  },
+  errorMessage: values.buildAddressErrorMessage,
+  inputs: [
+    {
+      id: "nonUKAddressApartmentNumber",
+      name: "nonUKAddressApartmentNumber",
+      label: { text: translate("fields.nonUKAddressApartmentNumber.label") },
+      classes: "govuk-input--width-5",
+      value: values.nonUKAddressApartmentNumber,
+      errorMessage: values.nonUKAddressApartmentNumberInValid
+    },
+    {
+      id: "nonUKAddressBuildingNumber",
+      name: "nonUKAddressBuildingNumber",
+      label: { text: translate("fields.nonUKAddressBuildingNumber.label") },
+      classes: "govuk-input--width-5",
+      value: values.nonUKAddressBuildingNumber,
+      errorMessage: values.nonUKAddressBuildingNumberInValid
+    },
+    {
+      id: "nonUKAddressBuildingName",
+      name: "nonUKAddressBuildingName",
+      label: { text: translate("fields.nonUKAddressBuildingName.label") },
+      classes: "govuk-input--width-20",
+      value: values.nonUKAddressBuildingName,
+      errorMessage: values.nonUKAddressBuildingNameInValid
+    }
+  ]
+} %}
+{{ customInput(args) }}
+{{ hmpoText(ctx, {
```

Comment:
It may be nicer to have the building address component as it's own component. Moving out inputs from streetName down into a different component or just into the enter-non-UK-address.njk file.